money.go

```diff
@@ -4,28 +4,59 @@ import (
 	"errors"
 )
 
-type amount struct {
-	val int
+// Amount is a datastructure that stores the amount being used for calculations
+type Amount struct {
+	val int64
 }
 
 // Money represents monetary value information, stores
 // currency and amount value
 type Money struct {
-	amount   *amount
+	amount   *Amount
 	currency *Currency
 }
 
 var calc *calculator
```

Comment:
This being a global doesn't currently make sense. A new one is made every time a new money object is made, but it just overwrites this global. As such, it seems like it will cause races.